gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -1699,4 +1815,52 @@ public static SnapshotListOption fields(SnapshotField... fields) {
    *     Deleting a snapshot</a>
    */
   Operation deleteSnapshot(String snapshot, OperationOption... options);
+
+  /**
+   * Creates a new image.
+   *
+   * @return a global operation for image's creation
+   * @throws ComputeException upon failure
+   */
+  Operation create(ImageInfo image, OperationOption... options);
+
+  /**
+   * Returns the requested image or {@code null} if not found.
+   *
+   * @throws ComputeException upon failure
+   */
+  Image get(ImageId imageId, ImageOption... options);
+
+  /**
+   * Lists images in the provided project that are available to the current user.
+   *
+   * @throws ComputeException upon failure
+   */
+  Page<Image> listImages(String project, ImageListOption... options);
```

Comment:
> What I think is more useful is to do what "gcloud compute images list" is doing and return all the publicly available images.

There is no such API. The only listing API for images requires a `projectId` to be passed.

> Also, do you think in this case we should return a functional Image where half of the operations will fail and the other half are doubtedly useful?

There is no easy way to distinguish between private and public images as `list(project)` could be also used passing the current project or a project that the user owns. The only possible check that comes to my mind is throwing when `image.imageId().project() != options.projectId()`, this would allow to keep `delete` and `deprecate` methods as is. However, this prevents (once `delete` and `deprecate` are fixed) the user from deleting image that he actually has rights to delete, only because they are in a different project. Thoughts?